packages/astro/src/assets/fonts/utils.ts

```diff
@@ -83,24 +92,10 @@ export type CacheHandler = ReturnType<typeof createCache>;
  * - `collect` will save the association of the original url and the new hash for later use
  * - the returned url will be `/_astro/fonts/<hash>.woff2`
  */
-export function createURLProxy({
-	hashString,
-	collect,
-}: {
-	hashString: (value: string) => string;
-	collect: (data: {
-		hash: string;
-		type: FontType;
-		value: string;
-	}) => string;
-}) {
-	return function proxyURL(value: string): string {
-		const type = extractFontType(value);
-		const hash = `${hashString(value)}.${type}`;
-		const url = collect({ hash, type, value });
-		// Now that we collected the original url, we return our proxy so the consumer can override it
-		return url;
-	};
+export function proxyURL(value: string, { hashString, collect }: ProxyURLOptions): string {
```

Comment:
nit: Options are usually optional, which isn't the case. Also, please start documenting stuff because it's getting very difficult to understand the flow of the code.

For example:
- the documentation of this method should be updated because `hashString` and `collect` aren't passed directly anymore, but they are part of the options
- `hashString` is a function that isn't documented anywhere and doesn't explain what it does.

You might want to use `JSDoc` to keep track of the variables.

Reply:
Updated and added a bunch of comments